src/main/java/org/apache/sysds/runtime/compress/lib/CLALibBinaryCellOp.java

```diff
@@ -104,11 +120,35 @@ public static MatrixBlock binaryOperationsLeft(BinaryOperator op, CompressedMatr
 	private static MatrixBlock binaryOperationsRightFiltered(BinaryOperator op, CompressedMatrixBlock m1,
 		MatrixBlock that) throws Exception {
 		BinaryAccessType atype = LibMatrixBincell.getBinaryAccessTypeExtended(m1, that);
-		if(that instanceof CompressedMatrixBlock && that.getInMemorySize() < m1.getInMemorySize()) {
+		if(that instanceof CompressedMatrixBlock){
```

Comment:
take the nested if statements, and refactor into a utility method named something appropriate.

Then make the nested statements into one, followed by another function call to a new method, that returns.


```java
if(newfunction(...)){
    return secondFunction(...);
}
```